### PR TITLE
fix wordpress-mobile/WordPress-Editor-Android#7: editor onboarding

### PR DESCRIPTION
**Pull request by maxme**
Fixes https://github.com/wordpress-mobile/WordPress-Editor-Android/issues/7

Message will appear on the Post List if the app is not a fresh install and if the user have visual editor enabled.

First idea was to show it on launch but I don't like how it can interfere if you open the app for specific purpose (read your notifications for instance).

I reused the Stats promo dialog style for consistency, and the same message shown in the iOS app.

To test:

* Do you get the dialog on app upgrade iff visual editor is enabled? (YES!)
* Do you get the dialog on a fresh install? (NO!)
* Make sure Stats dialog is correctly displayed.

Needs review: @aforcier 

screenshot-2016-03-04_09 30 37 522

### CODE REVIEW

**Comment by maxme** [comment]
> Message will appear on the Post List if the app is not a fresh install and if the user have visual editor enabled.
> 
> First idea was to show it on launch but I don't like how it can interfere if you open the app for specific purpose (read your notifications for instance).

Hum, since there is a New Post FAB on the My Sites screen, we should move the dialog there. So there is less risk of showing the dialog after the user was on the new editor. (that still can happen via Quick Post and Share with WP, and we could show the dialog at this moment, IMO not worth the complexity).

**Comment by aforcier** [comment]
Part of the text is cut off on my Nexus 5X:

7-editor-onboarding-text-cut-off

**Comment by nbradbury** [comment]
> Part of the text is cut off on my Nexus 5X:

I remember running into that with a similar dialog in stats. Switching from a LinearLayout to a RelativeLayout or FrameLayout should fix it.

**Comment by aforcier** [comment]
* [ ] Clean install of vanilla and alpha flavors shouldn't show editor dialog
* [ ] Upgrading `vanilla` > `vanilla` should not show editor dialog
* [ ] Upgrading `vanilla` or `zalpha` > `zalpha` should show dialog when visiting My Sites screen (tested upgrading while on the My Sites screen and on the reader screen)
* [ ] Stats dialog should be displaying correctly

File: WordPress/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -871,6 +871,10 @@
     <string name="editor_page_title_placeholder">Page Title</string>
     <string name="editor_content_placeholder">Share your story here…</string>
     <string name="visual_editor_enabled">Visual Editor enabled</string>
+    <string name="new_editor_promo_button_label">Great, Thanks!</string>
```
Comment (aforcier) [inline WordPress/src/main/res/values/strings.xml]: We should make 'Thanks' lowercase like iOS.

**Comment by aforcier** [comment]
Looks good except for the two things I mentioned @maxme, I'll test on older APIs once the layout issue is fixed.

**Comment by maxme** [comment]
Needs another round.

**Comment by aforcier** [comment]
Looks good on the 5X and on old APIs. :shipit:

**Pull request merged**